tests/sqlite_extension_test/sqlite_extension_test.ts

```diff
@@ -0,0 +1,88 @@
+// Copyright 2018-2025 the Deno authors. MIT license.
+import { DatabaseSync } from "node:sqlite";
+import { assertEquals, assertThrows } from "@std/assert";
+import * as path from "node:path";
+
+const extensionPath = (() => {
+  const isWindows = Deno.build.os === "windows";
+  const isMac = Deno.build.os === "darwin";
+  const isLinux = Deno.build.os === "linux";
+
+  const currentDir = new URL(".", import.meta.url).pathname;
+  const denoDir = path.resolve(currentDir, "..");
+
+  let libPrefix = "";
+  let libSuffix = "";
+
+  if (isWindows) {
+    libSuffix = "dll";
+  } else if (isMac) {
+    libPrefix = "lib";
+    libSuffix = "dylib";
+  } else if (isLinux) {
+    libPrefix = "lib";
+    libSuffix = "so";
+  } else {
+    throw new Error("Unsupported platform");
+  }
+
+  const targetDir = path.join(denoDir, "target", "debug");
+  return path.join(targetDir, `${libPrefix}test_sqlite_extension.${libSuffix}`);
+})();
+
+Deno.test({
+  name: "[node/sqlite] DatabaseSync loadExtension",
+  permissions: { read: true, write: true, ffi: true },
+  fn() {
+    // skip the test if the extension is not found
+    try {
+      Deno.statSync(extensionPath);
```

Comment:
Won't this cause the test to always succeed if something goes wrong?

Reply:
This is like `tests/ffi` so not ran directly using `deno test`. With `cargo t -p sqlite_extension_test` the extension is always built.